src/Reveal.Sdk.Dom.Tests/Data/DataSources/RestDataSourceFixture.cs

```diff
@@ -104,5 +109,70 @@ public void Method_SetsAndGetsValue_WithDifferentInputs(string method)
             Assert.Equal(method, dataSource.Method);
             Assert.Equal(method, dataSource.Properties.GetValue<string>("Method"));
         }
+
+        [Fact]
+        public void RDashDocument_HasCorrectDataSource_WhenLoadFromFile()
+        {
+            // Arrange
+            var filePath = Path.Combine(Environment.CurrentDirectory, "Dashboards", "TestRest.rdash");
+
+            // Act
+            var document = RdashDocument.Load(filePath);
+            var dataSource = document.DataSources.LastOrDefault();
+
+            // Assert
+            Assert.Equal(DataSourceProvider.REST, dataSource.Provider);
+            Assert.NotNull(dataSource.Properties.GetValue<string>("Url"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUseAnonymousAuthentication"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUsePreemptiveAuthentication"));
+        }
+
+        [Fact]
+        public void ToJsonString_CreatesFormattedJson_ForRestDataSource()
+        {
+            // Arrange
+            var expectedJson = @"
+            {
+              ""_type"": ""DataSourceType"",
+              ""Id"": ""Rest"",
+              ""Provider"": ""REST"",
+              ""Description"": ""Rest DS"",
+              ""Subtitle"": ""Excel2Json"",
+              ""Properties"": {}
+            }";
+
+            var dataSource = new RestDataSource()
```

Comment:
This is an internal object and we should be using it to test the public API. The hope is that maybe we can get away from even having this object.